### DISCUSSION TITLE
Are the runners for macOS really stuck on versions of Stack < 2.13.1?

### DISCUSSION THREAD

**Discussion by mpilgrem** [actions/virtual-environments#8494] [Installed software]
According to the readme files, the GitHub-hosted runner for `macos-11` has Stack 2.9.3, the runner for `macos-12` has Stack 2.9.3 and the runner for `macos-13` has Stack 2.11.1. The Linux and Windows runners have moved to Stack 2.13.1 (being the latest release of Stack).

Are the readme files accurate? If they are, is there something blocking the adoption of the latest release of Stack for macOS/x86_64 on those runners? (If they are not accurate, I'll raise a pull request.)

**Comment by vpolikarpov-akvelon** [comment]
Hey @mpilgrem. Stack is primarily installed by ghcup. The latest version of stack [supported](https://www.haskell.org/ghcup/install/#supported-tools) by ghcup is 2.11.1.

You can also use [haskell/actions/setup](https://github.com/haskell/actions/tree/main/setup) to select other version. By the way, the latest version (2.13.1 currently) is in the cache so it won't take long to install.

**Comment by mpilgrem** [comment]
@vpolikarpov-akvelon, thanks but my question is: are the readme files accurate and, if they are, why is macOS stuck on earlier versions of Stack when the Linux and Windows runners are not? If the macOS runners are using GHCup to install Stack, be aware that GHCup can install Stack 2.13.1, which I released a couple of weeks ago.

EDIT1: However, if `images/macos/provision/core/stack.sh` is relevant, it does not look to me like the runner is using GHCup to install Stack. EDIT2: From the perspective of the Stack project (which uses GitHub Actions for its CI), it would be better if the runner-supplied Stack was not installed by GHCup, because GHCup interferes with Stack's ability to upgrade itself (which can be used by the CI).

**Reply by vpolikarpov-akvelon** [reply]
Readme files are accurate, they are generated automatically after image is built.

Speaking of stack version itself and stack.sh script you mentioned, I didn't found any reasons for why stack on macOS is installed this way. It seems like nobody noticed that it is outdated (till now).

I've prepared PR to install stack the same way as we install it on Linux: https://github.com/actions/runner-images/pull/8525. Let's see whether it will work.

**Reply by vpolikarpov-akvelon** [reply]
Hey @mpilgrem. My [PR](https://github.com/actions/runner-images/pull/8525) have been merged and will be included in one of the closest updates.

**Reply by vpolikarpov-akvelon** [reply]
Hey @mpilgrem. Mac OS image updates have been deployed finally. All images now include Stack 2.13.1

**Reply by vpolikarpov-akvelon** [reply]
I'd like to warn you that Stack [has been removed](https://github.com/actions/runner-images/pull/8775) from Mac OS 13 images and won't be included to Mac OS 14 images as well.
Mac OS 13 update rollout is underway: https://github.com/actions/runner-images/pull/8821
